src/frontend/hooks/blog-form.js

```diff
@@ -0,0 +1,201 @@
+    setShowModal(true);
+  };
+
+  const closeModal = () => {
+    setShowModal(false);
+  };
+
+  const closeForm = useCallback(() => {
+    history.push('/blog-posts');
+  }, [history]);
+
+  const allFieldsAreFilled = () => {
+    let tmpErrors = []
+
+    // define errors that will be sent to the user
+    if (blogTitle === "") { tmpErrors.push("title cannot be blank") }
+    if (author === "") { tmpErrors.push("author cannot be blank") }
+    if (summary === "") { tmpErrors.push("summary cannot be blank") }
+    if (link === "") { tmpErrors.push("link cannot be blank") }
+    if (date === "") { tmpErrors.push("date cannot be blank") }
+    if (category === "") {  tmpErrors.push("category cannot be blank") }
+
+    if (currentImageURL == '' && uploadedImageURL == '') {
+          tmpErrors.push( "One image is required")
+        }
+
+    const pattern = /^(([0-9])|([0-2][0-9])|([3][0-1]))\-(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)\-\d{4}$/;
+    if (!pattern.test(date)){
+      tmpErrors.push("date must be in day-month-year form")
+    }
+
+    if (summary.length > 200 || summary.length < 1){
+      tmpErrors.push(`Summary must be between 1-200 characters, currently: ${summary.length}`)
+    }
+
+    setErrors(tmpErrors);
+
+    if (tmpErrors.length == 0) {
+      return true; // checks passed
+    }
+    return false // cheks failed
+  }
+
+  // Begin submitting form
+  const saveForm = useCallback(async () => {
+    // TODO: Still need to validate whether someone uploaded a file that already exists in google storage
+    if(!allFieldsAreFilled()) return;
+
+    try {
+      if (uploadedImageURL != '') {
+        // Send new image to google storage
+        const file = uploadedImageURL;
+
+        const data = new FormData();
+        data.append('files', file, file.name);
+
+        const res = await api.formUpload(data);
+
+        const imageUrls = res.data.data;
+        if (imageUrls.length === 0) {
+          throw new Error('Failed to upload image.');
+        }
+      }
+
+      // Prepare image url to store in DB
+      const googleStorageImageLink = "https://storage.googleapis.com/waterloop_cms_image_upload/"
+       // Either keep the current image or save the name of the uploaded file (date-name)
+      const imageToUpload = uploadedImageURL == '' ? currentImageURL : `${googleStorageImageLink}${Date.now()}-${uploadedImageURL.name.replaceAll(' ', '-')}`
+
+      const blogData = {
+        title: blogTitle,
+        author,
+        summary,
+        date,
+        link,
+        closed,
+        visibility,
+        category,
+        image: imageToUpload
+      };
+
+      // Send to DB
+      if (allFieldsAreFilled()) {
+        if (params.blogId) {
+          // update an EXISTING blog (by id) with the new blogData we just defined
+          const res = await api.blogs.editBlog(params.blogId, blogData);
+          const data = await res.data
+          console.log("Tried to edit blog, status: " , data)
```

Comment:
Can we just do `console.log("Tried to edit blog, status: " , res.data)` instead ? Is there a need for the second `await` ?

Reply:
Yeah, I think you're right.
I think I just put the extra await out of habit since I'm used to [converting the data to JSON which requires a second await](https://dmitripavlutin.com/javascript-fetch-async-await/), but I don't think that's necessary here, so I'll remove it